controllers/confidential_handler.go

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func patchConfigMap(client client.Client, name string, namespace string, data map[string]string) error {
+	configMap := &corev1.ConfigMap{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return err
+	}
+
+	for key, value := range data {
+		configMap.Data[key] = value
+	}
+
+	return client.Update(context.TODO(), configMap)
+}
+
+func (r *KataConfigOpenShiftReconciler) toggleFeatureConfidential(enabled bool) error {
+
+	var ImageConfigMapData map[string]string
+	var peerpodsCMData map[string]string
+
+	if enabled {
+		ImageConfigMapData = map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
+		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+	} else {
+		ImageConfigMapData = map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "no"}
+		peerpodsCMData = map[string]string{"DISABLECVM": "true"}
+	}
+
+	// Changes to make before image creation job starts:
+	//   - Edit ImageConfigMap to create a confidential image.
+
+	ig := GetImageGenerator()
+
+	if enabled {
+		// Create ImageConfigMap, if it doesn't exist already.
+		if err := ig.createImageConfigMapFromFile(); err != nil {
+			return err
+		}
+	}
+
+	// Patch ImageConfigMap.
+	if err := patchConfigMap(r.Client, ig.getImageConfigMapName(), OperatorNamespace, ImageConfigMapData); err != nil {
+		if !enabled && k8serrors.IsNotFound(err) {
+			// Nothing to do, feature is disabled and configMap doesn't exist.
+		} else {
+			return err
+		}
+	}
+
+	// Changes to make before pods start:
+	//   - Edit peer pods config to enable confidential.
+
+	// Patch peer pods configMap, if it exists.
+	if err := patchConfigMap(r.Client, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Should the configmap values be switched to default when this fg is disabled (as part of kataconfig deletion)?

Reply:
Also, I think since any changes to the image configmap values are only applicable before ImageCreate, other than resetting the configmap values on disabling of this fg, there is nothing else to be done.

Reply:
Frankly, I don't understand this function. There are subtle differences in the handing of each config map and it is hard to guess the outcome depending when it is called.

@spotlesstofu please explain the flow and expected result in the commit description, including any special situation (e.g. why is it fine to lack a config map or not).